Mac setup

## In which file did you encounter the issue?

MAC_SETUP.md

### Did you change the file? If so, how?

No

## Describe the issue

These instructions don't work on Catalina (10.15.2). If someone knows of a workaround, it would be great to add it here. Otherwise, we should add an acknowledgement so that users don't waste their time.

When I try to install pyenv, I get this message:
```
$ brew install pyenv
Warning: You are using macOS 10.15.
We do not provide support for this pre-release version.
You will encounter build failures with some formulae.
```

This time, I didn't get any warnings about Catalina being unsupported when installing pyenv, but I still can't install any versions of python.

```
$ brew --version
Homebrew 2.2.2
Homebrew/homebrew-core (git revision dd2f; last commit 2020-01-08)
$ pyenv install 2.7.17
python-build: use openssl from homebrew
python-build: use readline from homebrew
Downloading Python-2.7.17.tar.xz...
-> https://www.python.org/ftp/python/2.7.17/Python-2.7.17.tar.xz
Installing Python-2.7.17...
python-build: use readline from homebrew
python-build: use zlib from xcode sdk

BUILD FAILED (OS X 10.15.2 using python-build 20180424)

Inspect or clean up the working tree at /var/folders/sh/dlw4l1p57d11l51pc4674z0r00h79z/T/python-build.20200108113039.43913
Results logged to /var/folders/sh/dlw4l1p57d11l51pc4674z0r00h79z/T/python-build.20200108113039.43913.log

Last 10 log lines:
checking size of long long... 0
checking for long double support... yes
checking size of long double... 0
checking for _Bool support... yes
checking size of _Bool... 0
checking for uintptr_t... yes
checking size of uintptr_t... configure: error: in `/var/folders/sh/dlw4l1p57d11l51pc4674z0r00h79z/T/python-build.20200108113039.43913/Python-2.7.17':
configure: error: cannot compute sizeof (uintptr_t)
See `config.log' for more details
make: *** No targets specified and no makefile found.  Stop.
$
$ pyenv install 3.7.6
python-build: use openssl@1.1 from homebrew
python-build: use readline from homebrew
Downloading Python-3.7.6.tar.xz...
-> https://www.python.org/ftp/python/3.7.6/Python-3.7.6.tar.xz
Installing Python-3.7.6...
python-build: use readline from homebrew
python-build: use zlib from xcode sdk

BUILD FAILED (OS X 10.15.2 using python-build 20180424)

Inspect or clean up the working tree at /var/folders/sh/dlw4l1p57d11l51pc4674z0r00h79z/T/python-build.20200108113546.47662
Results logged to /var/folders/sh/dlw4l1p57d11l51pc4674z0r00h79z/T/python-build.20200108113546.47662.log

Last 10 log lines:
Parser/tokenizer.c:214:22: warning: this function declaration is not a prototype [-Wstrict-prototypes]
            buf[i] = tolower(c);
                     ^
./Include/pyport.h:591:29: note: expanded from macro 'tolower'
#define tolower(c) towlower(btowc(c))
                            ^
make: *** [Parser/acceler.o] Error 1
make: *** [Parser/listnode.o] Error 1
221 warnings and 11 errors generated.
make: *** [Parser/tokenizer.o] Error 1
$
```

--- a/MAC_SETUP.md
+++ b/MAC_SETUP.md
@@ -94,7 +94,7 @@ test their code.
 
     ```console
     $ which python
-    /Users/tswast/.pyenv/shims/python
+    ~/.pyenv/shims/python
     ```
 
 

--- a/MAC_SETUP.md
+++ b/MAC_SETUP.md
@@ -10,15 +10,18 @@ test their code.
 
 ## Before you begin
 
-1.  [Optional] Install [homebrew](https://brew.sh/).
+1. Install [homebrew](https://brew.sh/) if you do not already have it.
+
+   **Note:** If you are running Catalina (MacOS 10.15.x), ensure that you have a
+   compatible version of Homebrew (2.1.13 or later). Running `brew update` on
+   Catalina does not always result in a compatible version, so uninstall and
+   reinstall homebrew, if necessary.
 
 ## Installing pyenv and pyenv-virtualenv
 
 1.  Install [pyenv](https://github.com/pyenv/pyenv).
 
-    I (tswast@) use [homebrew](https://brew.sh/) to install it.
-
-    ```
+    ```console
     brew update
     brew install pyenv
     ```
@@ -26,7 +29,7 @@ test their code.
 1.  Install the [pyenv-virtualenv](https://github.com/pyenv/pyenv-virtualenv)
     plugin.
 
-    ```
+    ```console
     brew install pyenv-virtualenv
     ```
 
@@ -37,20 +40,12 @@ test their code.
     eval "$(pyenv virtualenv-init -)"
     ```
 
-    Note that this also works with ZSH.
+    **Note:** This also works with ZSH.
 
 1.  Reload your shell.
 
-    ```
+    ```console
     source ~/.bashrc
-    ```
-
-1.  After you have installed a python version through pyenv (see below), 
-    verify that you are now using the pyenv Python shim.
-
-    ```
-    $ which python
-    /Users/tswast/.pyenv/shims/python
     ```
 
 ## Installing multiple Python versions
@@ -58,59 +53,72 @@ test their code.
 
 1.  See the available Python versions with
 
-    ```
+    ```console
     pyenv install --list
     ```
 
-    The Python versions are at the top of the long list. If the Python
+    **Note:** The Python versions are at the top of the long list. If the Python
     version you want isn't listed, you may need to upgrade your pyenv with
     homebrew.
 
-    ```
+    ```console
     brew update
     brew upgrade pyenv
     ```
     
-    View the options related to a specific release with
-    
-    ```
-    pyenv install 3.x
-    ```
-    
-    
+1.  Install the necessary Python versions with pyenv. Use the latest release
+    of the versions you wish to test against.  A list of available versions
+    is available on [python.org](https://www.python.org/doc/versions/)
 
-1. Compile the necessary Python versions with pyenv. Use the latest release
-   of the versions you wish to test against.  A list of available versions
-   is available on [python.org](https://www.python.org/doc/versions/)
-
-    As of December 9, 2019 latest Python versions are:
+    As of January 8, 2020, the latest Python versions are:
 
     *  2.7.17 (latest 2.7.x release)
-    *  3.5.8 (latest 3.5.x release)
-    *  3.6.9 (latest 3.6.x release)
-    *  3.7.5 (latest 3.7.x release)
+    ```console
+    $ pyenv install 2.7.17
+    ```
+    *  3.5.9 (latest 3.5.x release)
+    ```console
+    $ pyenv install 3.5.9
+    ```
+    *  3.6.10 (latest 3.6.x release)
+    ```console
+    $ pyenv install 3.6.10
+    ```
+    *  3.7.6 (latest 3.7.x release)
+    ```console
+    $ pyenv install 3.7.6
+    ```
+
+1.  After you have installed a python version through pyenv,
+    verify that you are now using the pyenv Python shim.
+
+    ```console
+    $ which python
+    /Users/tswast/.pyenv/shims/python
+    ```
+
 
 ## Using pyenv and pyenv-virtualenv to manage your Python versions
 
 1.  Change to the desired source directory.
 
-    ```
+    ```console
     cd ~/src/python-docs-samples
     ```
 
 1.  Create a virtualenv using `pyenv virtualenv`.
 
-    ```
-    pyenv virtualenv 3.6.4 python-docs-samples
+    ```console
+    pyenv virtualenv 3.7.6 python-docs-samples
     ```
 
     This creates a virtualenv folder within `~/.pyenv/versions/`.
 
 1.  Set the local Python version(s) with `pyenv local`
 
-    ```
+    ```console
     # pyenv local [name of virtualenv] [list of python versions to use]
-    pyenv local python-docs-samples 3.6.4 3.7.0 3.5.4 2.7.15
+    pyenv local python-docs-samples 3.6.10 3.7.6 3.5.9 2.7.17
     ```
 
 1.  Now, when you `cd` into the source directory or a subdirectory within it,
@@ -120,4 +128,4 @@ test their code.
 
 1.  Add `.python-version` to your [global gitignore
     file](https://help.github.com/articles/ignoring-files/#create-a-global-gitignore),
-    so it wont be committed into the repository.
+    so it won't be committed into the repository.

